src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private int size;
+    private Node<T> current;
 
+    public void push(T value) {
+        if (size == 0) {
+            current = new Node<>(null, value);
+        } else {
+            Node<T> current = this.current;
+            this.current = new Node<>(current, value);
+        }
```

Comment:
```suggestion
        current = new Node<>(current, value);
```

Reply:
agree and let's check the size before this

Reply:
Why do we need to check the size here?